dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorPower.java

```diff
@@ -193,6 +194,7 @@ public List<Event> visitAtomicFetchOp(AtomicFetchOp e) {
         // Power does not have mo tags, thus we use null
         Load load = newRMWLoadExclusive(resultRegister, address, null);
         Store store = newRMWStoreExclusive(address, dummyReg, null, true);
+        store.addFilters(Tag.MATCHADDRESS);
```

Comment:
I think it is better to add `EventFactory.Power.newStoreConditional` which would return a `RMWStoreExclusive` with the appropriate tag. Obviously, RISCV should have its own version as well, which can even set the `STCOND` tag (if you need it for something).